modules/tensor_mechanics/doc/content/source/materials/FiniteStrainUObasedCP.md

```diff
@@ -1,11 +1,158 @@
+
++Function `computeQpStress`+:
+
+Input:
+
+- 2$^{nd}$ Piola-Kirchhoff (PK2) stress in the intermediate configuration at previous increment (variable `_pk2_old`, symbol $T_n$);
+- Plastic deformation gradient at previous increment (variable `_fp_old`, symbol $F^p_n$);
+- State variables at previous increment;
+- Current deformation gradient (variable `_dfgrd`, symbol $F$).
+
+Output:
+
+- Current PK2 stress (variable `_pk2`, symbol $T$);
+- Current plastic deformation gradient (variable `_fp`, symbol $F^p$);
+- Current slip system resistances.
+- State variables are solved iteratively using a predictor corrector algorithm. $T$ is solved using Newton Raphson.
+
++Flowchart+:
+
+i: At iteration i=0, $y^\alpha_i=y^\alpha_n$
+
+ii: Calculate the residual r and Jacobian J=$\partial r/\partial T$ from the function `calc_resid_jacob`
+
+iii: Update T as: $T_{i+1}=T_{i}-J^{-1}r$
+
+iv: Check $||r||_2 <$ `_rtol` (Optional user defined parameter)
+
+If False then go to Step ii
+
+Else
+
+v: Calculate $y^\alpha_{i+1}$ from function `updateSlipSystemResistanceAndStateVariable`
+
+vi: Check $max|s^\alpha_{i+1}-s^\alpha_{i}|/|s^\alpha_{i}| <$ `_stol`  (Optional user defined parameter)
+
+If False then go to Step ii
+
+Else
+
+vii: Obtain rotation tensor R
+
+viii: Update rotation as $\tilde{R}=RQ$ where Q is the rotation tensor from Euler angles
+
+Exit Function
+
++Function `calc_resid_jacob`+:
+User can override this function.
+
+Input:
+
+- PK2 stress at previous iteration (variable `_pk2`, symbol $T_i$, i - iteration);
+- Inverse of plastic deformation gradient at previous increment (variable `_fp_old_inv`, symbol $F^{p-1}_n$);
+- Current deformation gradient (variable `_dfgrd`, symbol $F$).
+
+Output:
+
+- Residual (variable `resid`, symbol r);
+- Jacobian (variable `jac`, symbol J);
+- Updated inverse of plastic deformation gradient (variable `_fp_inv`, symbol $F^{p-1}$)
+
++Residual r as implemented+:
+
+i. Get slip rates from userobject.
+
+ii. Resultant slip increment (variable `eqv_slip_incr`) $\Delta \gamma_{res}=I-\sum\limits_m\sum
+\limits_\alpha \left( \Delta \gamma^\alpha_m S_m^\alpha\right)$ where $S_m^\alpha$ is the flow direction.
+
+iii. Current plastic component of deformation gradient $F^{p-1}=F^{p-1}_n\Delta \gamma_{res}$.
+
+iv. Elastic component of deformation gradient in iteration i+1, $F^e=FF^{p-1}$.
+
+vi: PK2 stress due to $T_i$ and associated slip increment in current iteration i+1 (variable `pk2_new`), $T^\prime=C:\frac{1}{2}\left(F^{eT}F^e-I\right)$.
+
+vii. Residual $r=T_i-T^\prime$.
+
++Jacobian J formulation+:
+
+i. $r=T_i-T^\prime$ and $J=\frac{\partial r}{\partial T_i}=I-\frac{\partial T^\prime}{\partial T_i}$.
+
+ii. $ T^\prime=C:E^e$ which gives $\frac{\partial T^\prime}{\partial T_i}=C: \frac{\partial E^e}{\partial T_i}$.
```

Comment:
```suggestion
ii. $T^\prime=C:E^e$ which gives $\frac{\partial T^\prime}{\partial T_i}=C: \frac{\partial E^e}{\partial T_i}$.
```

Reply:
This one isn't rendering correctly.  I assume it's because of the space in front of T.